### ISSUE TITLE
Feature Request: Place HTML into templates for easy overriding

### ISSUE THREAD

**Issue by angusdowling** [advancedforms/advanced-forms#28]
## Need
I noticed in advanced-forms/core/core-forms.php that there's a lot of hardcoded HTML when the form gets rendered.

I think it would be a good idea to separate this HTML, moving it away from the logic and into multiple template files. These template files can be included in a way where they can be overridden. 

An example of this can be found by taking a look at how woocommerce implements their templating.

## Value
Overriding the HTML means the user has full control over what gets displayed on their page, including classes and whether or not they want to include further markup.

If you think this would be a good addition, I would be happy to do this and submit a PR.

**Comment by Fabianlindfors** [comment]
Hi!

This is a nice suggestion and I definitely want to make customization as easy as possible. One thing that worries me is overriden template files becoming outdated which seems to be a very common problem with Woocommerce. Preferably I would like to enable customization through hooks as far as possible although template files might be a necessary evil.

My focus is currently elsewhere in regards to Advanced Forms but if you want to have a go at it, feel free. I can't guarantee that I'll have the time to check it out or that I will judge it a good fit for the plugin right now but any contribution is very welcome. Thank you!

**Comment by angusdowling** [comment]
Cool, no worries. I'll get started on a potential solution and open a pull request.

**Comment by hacknug** [comment]
Did any of you guys started working on this?

**Comment by fabianlindfors** [comment]
No, unfortunately. There are a few things higher on my list right now (such as acf-json like functionality) but I'm hoping to look into this sooner or later.

If I may ask, what customizations are you aiming to add that can't be achieved right now? I might potentially be able to add some extra hooks for you to use.

Thanks!

**Comment by hacknug** [comment]
The most important thing would be adding custom classes to each element depending on its type (using TailwindCSS).

Second would be changing the markup structure to customize things like radios and checkboxes.

**Comment by fabianlindfors** [comment]
Right now you can add classes to the field wrappers using the `af/form/field_attributes` filter (https://advancedforms.github.io/filters/af-form-field_attributes/). Adding classes to the actual inputs is unfortunately not possible through my plugin as these are provided by ACF. Fortunately for you, ACF actually a way to add classes to inputs! This can be done by passing a class attribute for a field (can't be done through the UI). Here's an example on how to add a class for a specific text field:

```
function add_text_field_class( $field ) {
  $field['class'] = 'CLASS';
  return $field;
}
add_filter( 'acf/prepare_field/name=FIELD_NAME', 'add_text_field_class', 10, 1 );
```

In the same way, changing the markup for radios and checkboxes wouldn't be possible through my plugin either as those are provided entirely by ACF and limited in customizations to what ACF can provide. Template customizations wouldn't help aiding this problem either, unfortunately.

**Comment by sintacs** [comment]
If you do this, please make the af-success message template based too. :)

**Comment by fabianlindfors** [comment]
I will! Until then I actually just added a filter for the success message with version 1.5.4. This might help you customize the message to your liking:

Check it out: https://github.com/advancedforms/advanced-forms/commit/cc6ade136992c8eafe52a59b50cd589c0baa8bb3

**Comment by sintacs** [comment]
This is great, thank you. But with this filter I can't change the class of the div? <div class="af-success">

**Comment by fabianlindfors** [comment]
The div can’t currently be customized but using the filter you could add your own wrapper div with a custom class!

Fabian Lindfors

**Comment by sintacs** [comment]
This works, thanks!
